lightkurve/periodogram.py

```diff
@@ -885,16 +885,16 @@ def from_lightcurve(lc, **kwargs):
         maximum_period = kwargs.pop("maximum_period", None)
         period = kwargs.pop("period", None)
         if minimum_period is None:
-            if 'period' in kwargs:
-                minimum_period = period.min()
-            else:
+            if period is None:
                 minimum_period = np.max([np.median(np.diff(lc.time)) * 4,
                                          np.max(duration) + np.median(np.diff(lc.time))])
-        if maximum_period is None:
-            if 'period' in kwargs:
-                maximum_period = period.max()
             else:
+                minimum_period = np.min(period)
```

Comment:
You should catch nans somewhere in period

Reply:
Good point!  I re-arranged and edited the user input validation steps now to catch this, and added it to the unit test (c6b0630).